src/main/java/com/amazon/sqs/javamessaging/AmazonSQSExtendedClient.java

```diff
@@ -616,23 +621,36 @@ public SendMessageBatchResponse sendMessageBatch(SendMessageBatchRequest sendMes
             return super.sendMessageBatch(sendMessageBatchRequest);
         }
 
-        List<SendMessageBatchRequestEntry> batchEntries = new ArrayList<>(sendMessageBatchRequest.entries().size());
+        List<SendMessageBatchRequestEntry> originalEntries = sendMessageBatchRequest.entries();
+        ArrayList<SendMessageBatchRequestEntry> alteredEntries = new ArrayList<>(originalEntries.size());
+        alteredEntries.addAll(originalEntries);
 
+        // Batch entry sizes order by size
+        List<Pair<Integer, Long>> entrySizes = IntStream.range(0, originalEntries.size())
+                .boxed()
+                .map(i -> Pair.of(i, sizeOf(originalEntries.get(i))))
+                .sorted((p1, p2) -> Long.compare(p2.right(), p1.right()))
```

Comment:
what's the purpose of sorting here?

Reply:
The overall idea here is that we want to offload the least amount of batch entries possible. Less AWS calls means less time spent on roundtrips. 
So we sort by size, and after that we start moving entries from largest to smallest, stoping at the moment `totalSize` fits the limit.
We only offload according to the sort order, and preserve original order of entries in resulting batch request, as it is important for fifo use-case.